docs/development/Building in Windows 10 or 11 with MSYS2.md

```diff
@@ -1,109 +1,101 @@
 # Building in Windows with MSYS2
-- This environment does not require installing WSL which may not be available or would get in the way of other virtualization and/or anti-cheat softwares. 
-- It is also much faster to install and get set up because of its small size(~2 GB download and ~3.12 GB total after building hex file).
-
+- This environment does not require installing WSL, which may not be available or would get in the way of other virtualization and/or anti-cheat software
+- It is also much faster to install and get set up because of its small size(~3.65 GB total after building hex file as of 6.0.0)
 ## Setting up the environment
-
-1. Download and install the latest MSYS2 release from https://repo.msys2.org/distrib/x86_64/
-    - Scroll all the way down for an executable
-    - Scroll halfway down for a self-extracting archive
-
+### Download and install MSYS2
+1. For 6.0.0, the last version that works is [20220603](https://repo.msys2.org/distrib/x86_64/msys2-x86_64-20220603.exe)
```

Comment:
Could you share what happens when you use a newer version?

Reply:
At the very last step, on old msys2 versions up to 20220603:
```Linking C executable ../../../../bin/MATEKH743.elf
Memory region         Used Size  Region Size  %age Used
           FLASH:        1368 B       128 KB      1.04%
    FLASH_CONFIG:          0 GB       128 KB      0.00%
          FLASH1:      748923 B      1792 KB     40.81%
        ITCM_RAM:       14136 B        64 KB     21.57%
        DTCM_RAM:        7504 B       128 KB      5.73%
             RAM:       88960 B       512 KB     16.97%
          D2_RAM:        5248 B       256 KB      2.00%
       MEMORY_B1:          0 GB         0 GB
Built target MATEKH743.elf
Built target MATEKH743
```
Since msys2 version 20220904:
```
Linking C executable ../../../../bin/MATEKH743.elf
make: the '-j' option requires a positive integer argument
Usage: make [options] [target] ...
Options:
  -b, -m                      Ignored for compatibility.
  -B, --always-make           Unconditionally make all targets.
  -C DIRECTORY, --directory=DIRECTORY
                              Change to DIRECTORY before doing anything.
  -d                          Print lots of debugging information.
  --debug[=FLAGS]             Print various types of debugging information.
  -e, --environment-overrides
                              Environment variables override makefiles.
  -E STRING, --eval=STRING    Evaluate STRING as a makefile statement.
  -f FILE, --file=FILE, --makefile=FILE
                              Read FILE as a makefile.
  -h, --help                  Print this message and exit.
  -i, --ignore-errors         Ignore errors from recipes.
  -I DIRECTORY, --include-dir=DIRECTORY
                              Search DIRECTORY for included makefiles.
  -j [N], --jobs[=N]          Allow N jobs at once; infinite jobs with no arg.
  --jobserver-style=STYLE     Select the style of jobserver to use.
  -k, --keep-going            Keep going when some targets can't be made.
  -l [N], --load-average[=N], --max-load[=N]
                              Don't start multiple jobs unless load is below N.
  -L, --check-symlink-times   Use the latest mtime between symlinks and target.
  -n, --just-print, --dry-run, --recon
                              Don't actually run any recipe; just print them.
  -o FILE, --old-file=FILE, --assume-old=FILE
                              Consider FILE to be very old and don't remake it.
  -O[TYPE], --output-sync[=TYPE]
                              Synchronize output of parallel jobs by TYPE.
  -p, --print-data-base       Print make's internal database.
  -q, --question              Run no recipe; exit status says if up to date.
  -r, --no-builtin-rules      Disable the built-in implicit rules.
  -R, --no-builtin-variables  Disable the built-in variable settings.
  --shuffle[={SEED|random|reverse|none}]
                              Perform shuffle of prerequisites and goals.
  -s, --silent, --quiet       Don't echo recipes.
  --no-silent                 Echo recipes (disable --silent mode).
  -S, --no-keep-going, --stop
                              Turns off -k.
  -t, --touch                 Touch targets instead of remaking them.
  --trace                     Print tracing information.
  -v, --version               Print the version number of make and exit.
  -w, --print-directory       Print the current directory.
  --no-print-directory        Turn off -w, even if it was turned on implicitly.
  -W FILE, --what-if=FILE, --new-file=FILE, --assume-new=FILE
                              Consider FILE to be infinitely new.
  --warn-undefined-variables  Warn when an undefined variable is referenced.

This program built for x86_64-pc-msys
Report bugs to <bug-make@gnu.org>
lto-wrapper.exe: fatal error: make returned 2 exit status
compilation terminated.
c:/workspace/xpack/xpack-arm-none-eabi-gcc-10.2.1-1.1/bin/../lib/gcc/arm-none-eabi/10.2.1/../../../../arm-none-eabi/bin/ld.exe: error: lto-wrapper failed
collect2.exe: error: ld returned 1 exit status
make[3]: *** [src/main/target/MATEKH743/CMakeFiles/MATEKH743.elf.dir/build.make:6089: bin/MATEKH743.elf] Error 1
make[2]: *** [CMakeFiles/Makefile2:49448: src/main/target/MATEKH743/CMakeFiles/MATEKH743.elf.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:49481: src/main/target/MATEKH743/CMakeFiles/MATEKH743.dir/rule] Error 2
make: *** [Makefile:21548: MATEKH743] Error 2
```

Reply:
At the very last step, on old msys2 versions up to 20220603:
```Linking C executable ../../../../bin/MATEKH743.elf
Memory region         Used Size  Region Size  %age Used
           FLASH:        1368 B       128 KB      1.04%
    FLASH_CONFIG:          0 GB       128 KB      0.00%
          FLASH1:      748923 B      1792 KB     40.81%
        ITCM_RAM:       14136 B        64 KB     21.57%
        DTCM_RAM:        7504 B       128 KB      5.73%
             RAM:       88960 B       512 KB     16.97%
          D2_RAM:        5248 B       256 KB      2.00%
       MEMORY_B1:          0 GB         0 GB
Built target MATEKH743.elf
Built target MATEKH743
```
Since msys2 version 20220904:
```
Linking C executable ../../../../bin/MATEKH743.elf
make: the '-j' option requires a positive integer argument
Usage: make [options] [target] ...
Options:
  -b, -m                      Ignored for compatibility.
  -B, --always-make           Unconditionally make all targets.
  -C DIRECTORY, --directory=DIRECTORY
                              Change to DIRECTORY before doing anything.
  -d                          Print lots of debugging information.
  --debug[=FLAGS]             Print various types of debugging information.
  -e, --environment-overrides
                              Environment variables override makefiles.
  -E STRING, --eval=STRING    Evaluate STRING as a makefile statement.
  -f FILE, --file=FILE, --makefile=FILE
                              Read FILE as a makefile.
  -h, --help                  Print this message and exit.
  -i, --ignore-errors         Ignore errors from recipes.
  -I DIRECTORY, --include-dir=DIRECTORY
                              Search DIRECTORY for included makefiles.
  -j [N], --jobs[=N]          Allow N jobs at once; infinite jobs with no arg.
  --jobserver-style=STYLE     Select the style of jobserver to use.
  -k, --keep-going            Keep going when some targets can't be made.
  -l [N], --load-average[=N], --max-load[=N]
                              Don't start multiple jobs unless load is below N.
  -L, --check-symlink-times   Use the latest mtime between symlinks and target.
  -n, --just-print, --dry-run, --recon
                              Don't actually run any recipe; just print them.
  -o FILE, --old-file=FILE, --assume-old=FILE
                              Consider FILE to be very old and don't remake it.
  -O[TYPE], --output-sync[=TYPE]
                              Synchronize output of parallel jobs by TYPE.
  -p, --print-data-base       Print make's internal database.
  -q, --question              Run no recipe; exit status says if up to date.
  -r, --no-builtin-rules      Disable the built-in implicit rules.
  -R, --no-builtin-variables  Disable the built-in variable settings.
  --shuffle[={SEED|random|reverse|none}]
                              Perform shuffle of prerequisites and goals.
  -s, --silent, --quiet       Don't echo recipes.
  --no-silent                 Echo recipes (disable --silent mode).
  -S, --no-keep-going, --stop
                              Turns off -k.
  -t, --touch                 Touch targets instead of remaking them.
  --trace                     Print tracing information.
  -v, --version               Print the version number of make and exit.
  -w, --print-directory       Print the current directory.
  --no-print-directory        Turn off -w, even if it was turned on implicitly.
  -W FILE, --what-if=FILE, --new-file=FILE, --assume-new=FILE
                              Consider FILE to be infinitely new.
  --warn-undefined-variables  Warn when an undefined variable is referenced.

This program built for x86_64-pc-msys
Report bugs to <bug-make@gnu.org>
lto-wrapper.exe: fatal error: make returned 2 exit status
compilation terminated.
c:/workspace/xpack/xpack-arm-none-eabi-gcc-10.2.1-1.1/bin/../lib/gcc/arm-none-eabi/10.2.1/../../../../arm-none-eabi/bin/ld.exe: error: lto-wrapper failed
collect2.exe: error: ld returned 1 exit status
make[3]: *** [src/main/target/MATEKH743/CMakeFiles/MATEKH743.elf.dir/build.make:6089: bin/MATEKH743.elf] Error 1
make[2]: *** [CMakeFiles/Makefile2:49448: src/main/target/MATEKH743/CMakeFiles/MATEKH743.elf.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:49481: src/main/target/MATEKH743/CMakeFiles/MATEKH743.dir/rule] Error 2
make: *** [Makefile:21548: MATEKH743] Error 2
```